### PR TITLE
aa/cdh: make agent-config path configurable by env

### PR DESCRIPTION
**Pull request by mkulke**
relates-to: https://github.com/confidential-containers/cloud-api-adaptor/issues/1637

Since peerpods will template the agent-config.toml with aa_kbc_params and /etc might be on a read-only volume, we need to make this path configurable.

### CODE REVIEW

**Comment by mkulke** [comment]
cc @stevenhorsman @bpradipt

File: attestation-agent/lib/src/token.rs [hunk 1]
```diff
@@ -73,15 +77,19 @@ pub(crate) async fn get_kbc_params_from_config_file() -> Result<String> {
...
-    let agent_config_str = fs::read_to_string("/etc/agent-config.toml")
+    // check env for KATA_AGENT_CONFIG_PATH, fall back to default path
+    let path: &String = KATA_AGENT_CONFIG_PATH.get_or_init(|| {
+        env::var("KATA_AGENT_CONFIG_PATH").unwrap_or_else(|_| "/etc/agent-config.toml".into())
+    });
+
+    let agent_config_str = fs::read_to_string(path)
         .await
-        .map_err(|e| anyhow!("Failed to read /etc/agent-config.toml file: {e}"))?;
+        .context(format!("Failed to read ${path}"))?;
```
Comment (Xynnn007) [inline attestation-agent/lib/src/token.rs:87]: We might not need `$` here?

File: attestation-agent/lib/src/token.rs [hunk 2]
```diff
@@ -73,15 +77,19 @@ pub(crate) async fn get_kbc_params_from_config_file() -> Result<String> {
         aa_kbc_params: Option<String>,
     }
 
-    // Hard-code agent config path to "/etc/agent-config.toml" as a workaround
-    let agent_config_str = fs::read_to_string("/etc/agent-config.toml")
+    // check env for KATA_AGENT_CONFIG_PATH, fall back to default path
+    let path: &String = KATA_AGENT_CONFIG_PATH.get_or_init(|| {
+        env::var("KATA_AGENT_CONFIG_PATH").unwrap_or_else(|_| "/etc/agent-config.toml".into())
```
Comment (Xynnn007) [inline attestation-agent/lib/src/token.rs:82]: do you think it is good to `info!` or `debug!` here to tell which agent config is used?

**Review by Xynnn007 — APPROVED** [review]
This lgtm. However we might want to let CDH to have its own launch configuration file to set `aa_params` etc. https://github.com/confidential-containers/guest-components/issues/431

In this way, both peerpod and kata-cc could share a same interface for CDH.

**Comment by mkulke** [comment]
> This lgtm. However we might want to let CDH to have its own launch configuration file to set `aa_params` etc. #431
> 
> In this way, both peerpod and kata-cc could share a same interface for CDH.

I think we want to look at a consolidated/consistent way of doing configuration in guest-components eventually, but I would wait for the init-data concept to be finalized to start with that. 

This is more like a band-aid for the moment, since a hardcoded path doesn't work with a setup in which /etc isn't writeable. kata-agent on CCv0 is frozen atm and will be for some months probably, env is a way to pass the configuration from kata-agent to CDH & AA without altering kata-agent itself.

File: attestation-agent/lib/src/token.rs (hunk 2 above)
Comment (mkulke) [reply attestation-agent/lib/src/token.rs:83]: yup, added 👍

**Review by fitzthum — APPROVED** [review]

**Pull request merged**